kernel/src/grant.rs

```diff
@@ -3,49 +3,20 @@
 use core::marker::PhantomData;
 use core::mem::{align_of, size_of};
 use core::ops::{Deref, DerefMut};
-use core::ptr::{slice_from_raw_parts_mut, write, NonNull};
+use core::ptr::{write, NonNull};
 
 use crate::callback::AppId;
 use crate::process::{Error, ProcessType};
 use crate::sched::Kernel;
 
-/// Region of process memory reserved for the kernel.
-pub struct Grant<T: Default> {
-    pub(crate) kernel: &'static Kernel,
-    grant_num: usize,
-    ptr: PhantomData<T>,
-}
-
-pub struct AppliedGrant<T> {
-    appid: AppId,
-    grant: NonNull<T>,
-    _phantom: PhantomData<T>,
-}
-
-impl<T> AppliedGrant<T> {
-    pub fn enter<F, R>(self, fun: F) -> R
-    where
-        F: FnOnce(&mut Owned<T>, &mut Allocator) -> R,
-        R: Copy,
-    {
-        let mut allocator = Allocator { appid: self.appid };
-        let mut root = Owned::new(self.grant, self.appid);
-        fun(&mut root, &mut allocator)
-    }
-}
-
-pub struct Allocator {
-    appid: AppId,
-}
-
-pub struct Owned<T: ?Sized> {
-    data: NonNull<T>,
+pub struct Borrowed<'a, T: 'a + ?Sized> {
```

Comment:
/// Type that indicates a grant region has been entered and borrowed.
/// This is passed to capsules when they try to enter a grant region.